### PR TITLE
feat(sdk): add metadata field for importer

### PR DESCRIPTION
**Pull request by ji-yaqi**

### CODE REVIEW

File: sdk/RELEASE.md [hunk 1]
```diff
@@ -69,7 +70,7 @@
 ## Documentation Updates
 
 * N/A
-* 
+*
```
Comment (chensun) [inline sdk/RELEASE.md:73]: nit: remove the extra star left from a previous change.

File: sdk/python/kfp/v2/compiler/pipeline_spec_builder.py [hunk 2]
```diff
@@ -357,6 +356,11 @@ def build_importer_spec_for_task(
         type_schema=type_schema,
         reimport=task.importer_spec.reimport)
 
+    if task.importer_spec.metadata and len(task.importer_spec.metadata) != 0:
```
Comment (chensun) [inline sdk/python/kfp/v2/compiler/pipeline_spec_builder.py:359]: nit: `len(task.importer_spec.metadata) != 0` is redundant, the first condition covers it already.

**Comment by rzillerueloatgoogle** [comment]
LGTM + 1.

**Comment by chensun** [comment]
> @ji-yaqi: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> kubeflow-pipelines-tfx-python37	[fa2f2c7](https://github.com/kubeflow/pipelines/commit/fa2f2c734a867c185aa1477708161bbcbf27c367)	[link](https://oss-prow.knative.dev/view/gs/oss-prow/pr-logs/pull/kubeflow_pipelines/7112/kubeflow-pipelines-tfx-python37/1478839808884215808)	true	`/test kubeflow-pipelines-tfx-python37`

https://github.com/kubeflow/pipelines/pull/7143 will fix the failing test.

**Comment by chensun** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**